### DISCUSSION TITLE
Feature request : Import, Dark mode switch, Auto layout, Constraints, Presets, Transitions and Overlay

### DISCUSSION THREAD

**Discussion by saraninvincible** [uxbox/uxbox#893] [Ideas]
Hi Team,

I'm an Experience Designer and I find this project amazing. Also, I'm a pro-level user of XD and Figma.
I have a few feedback that could make penpot better.
1. **Import XD, Figma & Sketch** files into Penpot and vice versa
2. **Dark / Light Mode switch in developer handoff** to view dark version of prototype with ease (Designer has to define the library as per that) --> No app has this. It is one major problem to be solved.
3. **Auto layout** / Stacking of elements (Horizontal and vertical stacking with padding)
4. **Constraints** for elements (Top, bottom, Left , Right, Center)
5. **Preset of Artboards** (iPhone, Android, iPad etc)
6. **Prototype transitions** (Slide up, Slide Down, Dissolve, Slide right, Slide Left)
7. **Overlay function** in prototype mode (Lets the artboard float on another - useful for reusing of components like modals & popups)

Do let me know if you need any help in testing out any feature or having doubts on how features should behave. I'm happy to help ! :)

**Comment by iwoithe** [comment]
> Preset of Artboards (iPhone, Android, iPad etc)

Correct me if I've misinterpreted what you said, but Penpot already has preset sizes (see screenshot below).

![penpot](https://user-images.githubusercontent.com/73422868/116967854-66b2f780-acf2-11eb-8d42-da58820723bc.png)

**Reply by saraninvincible** [reply]
Yes, you are correct. That’s the one.
May I know the steps to reproduce it?
Coz I was unable to find the list of presets when I hit the ‘New artboard’ icon so I could proceed selecting the artboard size which will create the relevant artboard.

**Reply by iwoithe** [reply]
1) Select the ``New Artboard`` icon and create a new artboard

2) Select the artboard and you will see the size presets dropdown

**Comment by niwinz** [comment]
Yes, many of the proposed features are already implemented or are already in development.
In any case, thank you very much for the feedback.

**Reply by saraninvincible** [reply]
Glad to hear that
Thanks! 😃

**Comment by myfunnyandy** [comment]
We've just released the [resizing constraints feature](https://help.penpot.app/user-guide/constraints/) and [import/export for Penpot files](https://help.penpot.app/user-guide/import-export/) (.fig and .sketch coming next).

**Comment by skpaul** [comment]
Please try to add a feature something like 'auto-layout' in figma.

**Reply by Alotor** [reply]
It's one of our main focuses right now, we hope to start development soon. Thanks for your support.

**Reply by kevinloustau** [reply]
Definitely waiting for this feature, to switch to Penpot 👍

**Reply by dick-jo** [reply]
@Alotor I posted a relatively in depth response to an autolayout enhancement request issue that I believe you should take into consideration: https://github.com/penpot/penpot/issues/2268#issuecomment-1248763815

Please feel free to direct me to a more suitable ticket for this one

**Comment by jcklpe** [comment]
some update on what's going on with this feature: https://community.penpot.app/t/what-are-penpot-plans-for-autolayout/112

**Reply by myfunnyandy** [reply]
Is under development and in a very advanded state, we'll hope to have news soon. Here's an [sneak peek](https://community.penpot.app/t/autolayout-sneak-peek/153) (from some time ago, is much more mature now).

**Comment by saraninvincible** [comment]
Thanks
I’ll check on it :)
